test/unit/range/decorator/gap_decorator_test.cpp

```diff
@@ -58,6 +57,70 @@ class aligned_sequence<gap_decorator<inner_type_>> : public ::testing::Test
 
 INSTANTIATE_TYPED_TEST_CASE_P(gap_decorator, aligned_sequence, test_types);
 
+template <>
+struct iterator_fixture<decorator_t::iterator> : public ::testing::Test
+{
+    using iterator_tag = std::bidirectional_iterator_tag;
+    static constexpr bool const_iterable = true;
+
+    std::vector<dna4> const vec{"ACTGACTG"_dna4};
+
+    template <typename t>
+    static void initialise_with_gaps(t & v)
+    {
+        insert_gap(v, std::next(v.begin(), 5), 4);
+        insert_gap(v, std::next(v.begin(), 2));
+        insert_gap(v, v.end(), 3);
+        insert_gap(v, v.begin(), 5);
+    }
+
+    std::vector<gapped<dna4>> expected_range = [this] ()
+        {
+            std::vector<gapped<dna4>> tmp{};
+            std::copy(vec.begin(), vec.end(), std::back_inserter(tmp));
+            initialise_with_gaps(tmp);
+            return tmp;
+        }();
+
+    decorator_t test_range = [this] ()
+        {
+            decorator_t tmp{vec};
+            initialise_with_gaps(tmp);
+            return tmp;
+        }();
+};
+
+template <>
+struct iterator_fixture<decorator_t::iterator_ra> : iterator_fixture<decorator_t::iterator>
+{
+    using iterator_tag = std::random_access_iterator_tag;
+    static constexpr bool const_iterable = true;
+
+    // expected_range is inherited
+
+    struct expose_random_access_iterator
+    {
+        std::vector<dna4> const vec{"ACTGACTG"_dna4};
+        decorator_t range = [this] ()
+            {
+                decorator_t tmp{vec};
+                iterator_fixture<decorator_t::iterator>::initialise_with_gaps(tmp);
+                return tmp;
+            }();
+
+        decorator_t::iterator_ra begin() { return range.begin_ra(); }
+        decorator_t::iterator_ra end()   { return range.end_ra(); }
+        decorator_t::const_iterator_ra cbegin(){ return range.cbegin_ra(); }
+        decorator_t::const_iterator_ra cend()  { return range.cend_ra(); }
+    };
+
+    expose_random_access_iterator test_range{};
+};
+
+using test_type = ::testing::Types<decorator_t::iterator, decorator_t::iterator_ra>;
+
+INSTANTIATE_TYPED_TEST_CASE_P(iterator_fixture, iterator_fixture, test_type);
```

Comment:
can you please give it a more meaning full name like gap_decorator_iterator instead of iterator_fixture?

Reply:
Did you revert your last changes? I see a lot of stuff that is not resolved but I thought in a previous version you already did? Either something is messing with my eyes or something strange happened. So is the instantiated test still called iterator_fixture in your source?

Reply:
something is messing with my PR o.O
I'll revisit all the review comments. I am sure I changed this already.